flink-formats/flink-avro-confluent-registry/src/main/java/org/apache/flink/formats/avro/registry/confluent/debezium/DebeziumAvroDeserializationSchema.java

```diff
@@ -0,0 +1,208 @@
+ * The deserialization schema knows Debezium's schema definition and can extract the database data
+ * and convert into {@link RowData} with {@link RowKind}.
+ * <p>
+ * <p>Deserializes a <code>byte[]</code> message as a JSON object and reads
+ * the specified fields.
+ * <p>
+ * <p>Failures during deserialization are forwarded as wrapped IOExceptions.
+ *
+ * @see <a href="https://debezium.io/">Debezium</a>
+ */
+@Internal
+public final class DebeziumAvroDeserializationSchema implements DeserializationSchema<RowData> {
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * snapshot read
+	 */
+	private static final String OP_READ = "r";
+	/**
+	 * insert operation
+	 */
+	private static final String OP_CREATE = "c";
+	/**
+	 * update operation
+	 */
+	private static final String OP_UPDATE = "u";
+	/**
+	 * delete operation
+	 */
+	private static final String OP_DELETE = "d";
+
+	private static final String REPLICA_IDENTITY_EXCEPTION = "The \"before\" field of %s message is null, " +
+		"if you are using Debezium Postgres Connector, " +
+		"please check the Postgres table has been set REPLICA IDENTITY to FULL level.";
+
+	/**
+	 * The deserializer to deserialize Debezium Avro data.
+	 */
+	private final AvroRowDataDeserializationSchema avroDeserializer;
+
+	/**
+	 * TypeInformation of the produced {@link RowData}.
+	 **/
+	private final TypeInformation<RowData> resultTypeInfo;
+
+	/**
+	 * Debezium Avro data rowType
+	 */
+	private final RowType rowType;
+
+
+	public DebeziumAvroDeserializationSchema(
+			RowType rowType,
+			TypeInformation<RowData> resultTypeInfo,
+			String schemaRegistryUrl) {
+		this.resultTypeInfo = resultTypeInfo;
+		this.rowType = rowType;
+		RowType debeziumAvroRowType = createDebeziumAvroRowType(fromLogicalToDataType(rowType));
+
+		this.avroDeserializer = new AvroRowDataDeserializationSchema(
+			ConfluentRegistryAvroDeserializationSchema.forGeneric(
+				AvroSchemaConverter.convertToSchema(debeziumAvroRowType),
+				schemaRegistryUrl),
+			AvroToRowDataConverters.createRowConverter(debeziumAvroRowType),
+			resultTypeInfo);
+	}
+
+	public DebeziumAvroDeserializationSchema(
+			RowType rowType,
+			TypeInformation<RowData> resultTypeInfo,
+			AvroRowDataDeserializationSchema avroDeserializer) {
+		this.rowType = rowType;
+		this.resultTypeInfo = resultTypeInfo;
+		this.avroDeserializer = avroDeserializer;
+	}
+
+	@Override
+	public RowData deserialize(byte[] message) throws IOException {
+		throw new RuntimeException(
+			"Please invoke DeserializationSchema#deserialize(byte[], Collector<RowData>) instead.");
+	}
+
+	@Override
+	public void deserialize(byte[] message, Collector<RowData> out) throws IOException {
+
+		if (message == null || message.length == 0) {
+			// skip tombstone messages
+			return;
+		}
+		try {
+			GenericRowData row = (GenericRowData) avroDeserializer.deserialize(message);
+
+			GenericRowData before = null;
+			int index = rowType.getFieldIndex("before");
```

Comment:
We should also know the index of before and after. Shouldn't get the index dynamically. 
See the implementation of debezium-json.